library/src/main/camera2/com/google/android/cameraview/Camera2.java

```diff
@@ -422,9 +422,19 @@ private void collectCameraInfo() {
             mPreviewSizes.add(new Size(size.getWidth(), size.getHeight()));
         }
         mPictureSizes.clear();
-        for (android.util.Size size : map.getOutputSizes(ImageFormat.JPEG)) {
-            mPictureSizes.add(new Size(size.getWidth(), size.getHeight()));
+        // try to get hi-res output sizes for Marshmellow and higher
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            for (android.util.Size size : map.getHighResolutionOutputSizes(ImageFormat.JPEG)) {
+                mPictureSizes.add(new Size(size.getWidth(), size.getHeight()));
+            }
+        }
+        // fallback camera sizes and lower than Marshmellow
+        if(mPictureSizes.ratios().size() == 0){
```

Comment:
No, but could you format only the lines that you edited? Sorry for not being clear.